lib/workers/repository/process/fetch.ts

```diff
@@ -60,102 +62,146 @@ async function fetchDepUpdates(
   depConfig.packageName ??= depConfig.depName;
+
   if (depConfig.ignoreDeps!.includes(depName!)) {
     // TODO: fix types (#22198)
     logger.debug(`Dependency: ${depName!}, is ignored`);
     dep.skipReason = 'ignored';
-  } else if (depConfig.enabled === false) {
+    return Result.ok(dep);
+  }
+
+  if (depConfig.enabled === false) {
     logger.debug(`Dependency: ${depName!}, is disabled`);
     dep.skipReason = 'disabled';
-  } else {
-    if (depConfig.datasource) {
-      const { val: updateResult, err } = await withLookupStats(
-        depConfig.datasource,
-        () =>
-          Result.wrap(lookupUpdates(depConfig as LookupUpdateConfig)).unwrap(),
-      );
-
-      if (updateResult) {
-        Object.assign(dep, updateResult);
-      } else {
-        if (
-          packageFileConfig.repoIsOnboarded === true ||
-          !(err instanceof ExternalHostError)
-        ) {
-          return Result.err(err);
-        }
-
-        const cause = err.err;
-        dep.warnings ??= [];
-        dep.warnings.push({
-          topic: 'Lookup Error',
-          // TODO: types (#22198)
-          message: `${depName!}: ${cause.message}`,
-        });
-      }
-    }
-    dep.updates ??= [];
+    return Result.ok(dep);
   }
-  return Result.ok(dep);
-}
 
-async function fetchManagerPackagerFileUpdates(
-  config: RenovateConfig,
-  managerConfig: RenovateConfig,
-  pFile: PackageFile,
-): Promise<void> {
-  const { packageFile } = pFile;
-  const packageFileConfig = mergeChildConfig(managerConfig, pFile);
-  if (pFile.extractedConstraints) {
-    packageFileConfig.constraints = {
-      ...pFile.extractedConstraints,
-      ...config.constraints,
-    };
+  if (!depConfig.datasource) {
+    dep.updates ??= [];
+    return Result.ok(dep);
   }
-  const { manager } = packageFileConfig;
-  const queue = pFile.deps.map(
-    (dep) => async (): Promise<PackageDependency> => {
-      const updates = await fetchDepUpdates(packageFileConfig, dep);
-      return updates.unwrapOrThrow();
-    },
-  );
-  logger.trace(
-    { manager, packageFile, queueLength: queue.length },
-    'fetchManagerPackagerFileUpdates starting with concurrency',
-  );
 
-  pFile.deps = await p.all(queue);
-  logger.trace({ packageFile }, 'fetchManagerPackagerFileUpdates finished');
+  logger.trace({ depName }, 'Dependency lookup start');
+  const start = Date.now();
+
+  const result = await Result.wrap(
+    lookupUpdates(depConfig as LookupUpdateConfig),
+  )
+    .onValue((dep) => {
+      logger.trace({ dep }, 'Dependency lookup success');
+    })
+    .onError((err) => {
+      logger.trace({ err, depName }, 'Dependency lookup error');
+    })
+    .catch((err): Result<UpdateResult, Error> => {
+      if (
+        packageFileConfig.repoIsOnboarded === true ||
+        !(err instanceof ExternalHostError)
+      ) {
+        return Result.err(err);
+      }
+
+      const cause = err.err;
+      return Result.ok({
+        updates: [],
+        warnings: [
```

Comment:
are we sure `warnings` was never initialized before? otherwise the flat merge below will overwrite existing values with this.

Reply:
Since it's the `catch` path, it's unlikely we've pushed any warnings before